src/groups_v2/operations.rs

```diff
@@ -84,6 +88,19 @@ impl GroupOperations {
         }
     }
 
+    fn decrypt_pni(
+        &self,
+        ciphertext: &[u8],
+    ) -> Result<Pni, GroupDecodingError> {
+        match self
+            .group_secret_params
+            .decrypt_service_id(bincode::deserialize(ciphertext)?)?
+        {
+            ServiceId::Pni(pni) => Ok(pni),
+            ServiceId::Aci(_aci) => Err(GroupDecodingError::NotPni),
```

Comment:
It could be useful to already throw a `tracing::error` in here, since this is behavior that shouldn't happen and probably indicates a bug somewhere.

I could imagine that a pni-only field gets a separate encoding (e.g. without any service-id indicator) some day, and that we don't catch this. The same probably goes for ACI.

Reply:
Done!